From c443a245c58862064f1530841f96e6cb46503243 Mon Sep 17 00:00:00 2001
From: Elliott Clark <eclark@apache.org>
Date: Wed, 30 Sep 2015 15:30:15 -0700
Subject: [PATCH] HBASE-14529 Respond to SIGHUP to reload config

Summary: Add a signal handler to reload the config.

Test Plan: Send the command and see that configs get reloaded.

Differential Revision: https://reviews.facebook.net/D47877
---
 .../java/org/apache/hadoop/hbase/regionserver/HRegionServer.java | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java
index 59d13fa..0de28e6 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java
@@ -186,6 +186,8 @@ import com.google.protobuf.RpcCallback;
 import com.google.protobuf.RpcController;
 import com.google.protobuf.Service;
 import com.google.protobuf.ServiceException;
+import sun.misc.Signal;
+import sun.misc.SignalHandler;
 
 /**
  * HRegionServer makes a set of HRegions available to clients. It checks in with
@@ -603,6 +605,13 @@ public class HRegionServer extends HasThread implements
     putUpWebUI();
     this.walRoller = new LogRoller(this, this);
     this.choreService = new ChoreService(getServerName().toString());
+
+    Signal.handle(new Signal("HUP"), new SignalHandler() {
+      public void handle(Signal signal) {
+        getConfiguration().reloadConfiguration();
+        configurationManager.notifyAllObservers(getConfiguration());
+      }
+    });
   }
 
   protected TableDescriptors getFsTableDescriptors() throws IOException {
-- 
2.5.3

